item 조회 기능 개발

GSB0203 opened the pull request:
## 🎫 관련 이슈
[//]: # (다음 키워드를 사용하면 해당 PR을 머지할 때 자동으로 이슈를 닫을 수 있습니다.)
[//]: # (keyword: close|closes|closed|resolve|resolves|resolved|fix|fixes|fixed)
[//]: # (예시: close #1)

close #14 

<br>

## 📄 개요
[//]: # (작업 내용을 간단히 요약해서 적습니다.)
[//]: # (예시: 유저 회원가입 기능을 만들었습니다.)
>아이템 조회 기능을 만들었습니다.
<br>

## 🔨 작업 내용
[//]: # (작업 내용을 자세하게 적습니다.)
[//]: # (붙임표 "-" 을 사용해서 목록을 만듭니다.)
[//]: # (예시: 유저 회원가입 API를 만들었습니다.)

- 아이템 조회 API를 만들었습니다.


<br>

## 🏁 확인 사항
[//]: # (PR을 보내기 전 다음 사항을 확인해주세요.)
[//]: # (해당 사항을 모두 이행해야 머지할 수 있습니다.)
[//]: # (- [x] 를 사용해서 완료로 표시할 수 있습니다.)

- [ ] 테스트를 완료했나요?
- [ ] API 문서를 작성했나요?
- [x] 코드 컨벤션을 준수했나요?
- [x] 불필요한 로그, 주석, import 등을 삭제했나요?

<br>

## 🙋🏻 덧붙일 말
[//]: # (다음 사항이 있다면 적어주세요.)
[//]: # (PR에 대한 추가 설명)
[//]: # (중점적으로 리뷰받고 싶은 부분)
[//]: # (기타 등등)

GSB0203 committed:
feat(Item): "Item Entity"

Item Entity 및 Category Enum 생성

GSB0203 committed:
feat(Item): "Item 조회 Api"

Item 조회에 필요한 로직 구현

YunChan-Oh reviewed (commented):
패키지 구조는 Item 패키지를 만들고 그 안에 계층별로 쪼개는 게 좋다고 생각합니다.

YunChan-Oh commented on src/main/java/com/space/server/core/domain/item/Item.java:
```diff
@@ -0,0 +1,41 @@
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  @Column(name = "item_id")
+  @NotNull
+  private Long id;
```
@Id와 @GeneratedValue 어노테이션을 사용하여서 @NotNull을 쓸 필요는 없다고 생각합니다.

YunChan-Oh commented on src/main/java/com/space/server/core/domain/item/Item.java:
```diff
@@ -0,0 +1,41 @@
+  private String name;
+
+  @NotNull
+  private int price;
```
유효성 검사를 하는 어노테이션을 추후 넣어도 좋다고 생각합니다.

YunChan-Oh commented on src/main/java/com/space/server/core/presentation/item/dto/response/ItemResponse.java:
```diff
@@ -0,0 +1,18 @@
+) {
+  public static ItemResponse of(Item item) {
+    return new ItemResponse(item.getId(), item.getName(), item.getPrice(), item.getImage(), item.getCategory());
+  }
```
매개변수를 한 개 받아서 반환해주는거라 from으로 메서드 명을 짓는 게 더 낫다고 생각합니다.

GSB0203 committed:
Rename(Item): "Item 폴더 구조 변경"

Item 폴더 구조를 더 보기 쉽게 바꾸고 불필요해진 gitkeep 파일을 삭제하였습니다.

GSB0203 committed:
chore(item): 코드 리뷰 반영

Item Entity 속 중복 Null 처리 제거, 현재 코드 구현 의미에 맞게 Null 처리 조건 제거, ItemResponse 반환 매서드의  매개변수가 1개이기 때문에 변경

GSB0203 committed:
Merge branch 'master' of https://github.com/SPACE-FOR-SPACE/SPACE-SERVER into feat/#14

GSB0203 committed:
chore(gradle): test 수정

CI 테스트를 통과하지 못하여 수정

GSB0203 committed:
chore(gradle): test 삭제

CI 테스트를 통과하지 못하여 삭제
